WallPaperDev/WallPaperDev/Views/Controllers/SearchImageViewController.swift

```diff
@@ -0,0 +1,147 @@
+    
+    navigationItem.hidesSearchBarWhenScrolling = false
+    navigationItem.searchController = searchController
+  }
+  
+  private func setupBlueButton() {
+    self.view.addSubview(createImageButton)
+    createImageButton.setTitle("Create", for: .normal)
+    createImageButton.addTarget(self, action: #selector(backToChooseImageVC), for: .touchUpInside)
+    NSLayoutConstraint.activate([
+      createImageButton.widthAnchor.constraint(equalTo: self.view.widthAnchor, multiplier: 0.6),
+      createImageButton.heightAnchor.constraint(equalToConstant: 50),
+      createImageButton.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor),
+      createImageButton.centerXAnchor.constraint(equalTo: self.view.centerXAnchor)
+    ])
+  }
+  
+  @objc private func backToChooseImageVC() {
+    if let image = viewModel.selectedImage {
+      selectedImageDelegate?.passImageSelected(image: image)
+    }
+    navigationController?.popViewController(animated: true)
+  }
+}
+
+// MARK: - CollectionView Delegate
+extension SearchImageViewController : UICollectionViewDelegate {
+  func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    guard let selectedCell = searchImagesCV.cellForItem(at: indexPath) as? SearchImagesCell else {
+      return
+    }
+    viewModel.selectedImage = selectedCell.cellImage
+  }
+}
+
+// MARK: - CollectionView DataSource
+extension SearchImageViewController : UICollectionViewDataSource {
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    viewModel.imageURLS.count
+  }
+
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    guard let cell = searchImagesCV.dequeueReusableCell(withReuseIdentifier: searchImagesCV.cellID, for: indexPath) as? SearchImagesCell else {
+      return UICollectionViewCell()
+    }
+
+    let urlString = viewModel.imageURLS[indexPath.row].urls.regular
+
+    viewModel.networkManager.getPhotoData(from: urlString) { result in
+      switch result {
+      case let .success(image):
+        DispatchQueue.main.async {
+          cell.getImage(image)
+        }
+
+      case let .failure(error):
+        #if DEBUG
+          print("Unable to get photo data for url: \(urlString)")
+          print(error)
+        #endif
+      }
+    }
+    return cell
+  }
+}
+// MARK: - SearchBar Delegate
+extension SearchImageViewController : UISearchBarDelegate {
+  func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+    // set urls to empty for new search
+    viewModel.imageURLS = []
+    let keywords = searchBar.text
+    guard let finalKeyword = keywords?.replacingOccurrences(of: " ", with: "+") else { return }
+    viewModel.loadSearchURLS(for: finalKeyword, cv: searchImagesCV)
```

Comment:
Let's consider passing the raw search term to the view model and then have it do the replacements